packages/hub/src/hub.ts

```diff
@@ -263,6 +263,22 @@ export class Hub implements HubInterface {
       return null;
     }
   }
+
+  /**
+   * @inheritDoc
+   */
+  public traceHeaders(): { [key: string]: string } {
+    const top = this.getStackTop();
+    if (top.scope && top.client) {
```

Comment:
I'd change this to

```js
const top = this.getStackTop();

if (!top.scope || !top.scope.getSpan()) return {};

return {
  'sentry-trace': top.scope.getSpan().toTraceparent(),
};
```

We don't use `client` here, so there's no need to check for it.

Reply:
After trying it, it is more complex because of how it's structured, will leave it like this.